From 33ee003d23e59321a0217671c8d8eec2e2a67af9 Mon Sep 17 00:00:00 2001
From: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Date: Tue, 26 Aug 2014 17:19:06 -0400
Subject: [input] alps: change decode function prototype to return an int

Message-id: <1409073549-31548-16-git-send-email-benjamin.tissoires@redhat.com>
Patchwork-id: 89251
O-Subject: [PATCH RHEL7 15/18] Input: alps - change decode function prototype to return an int
Bugzilla: 1107819
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1107819
Upstream: since v3.17
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7882179
Tested by me on a Toshiba Portege R30 with a v7 Alps touchpad

commit 38c11eaaab0cf8ef6004aa704f1bb2ff5e6bc1b0
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Fri Jul 25 22:48:44 2014 -0700

    Input: alps - change decode function prototype to return an int

    So that decode functions can return a failure when appropriate.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>

Signed-off-by: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/input/mouse/alps.c b/drivers/input/mouse/alps.c
index 5026600..8f79491 100644
--- a/drivers/input/mouse/alps.c
+++ b/drivers/input/mouse/alps.c
@@ -538,7 +538,7 @@ static void alps_decode_buttons_v3(struct alps_fields *f, unsigned char *p)
  f->ts_middle = !!(p[3] & 0x40);
 }
 
-static void alps_decode_pinnacle(struct alps_fields *f, unsigned char *p,
+static int alps_decode_pinnacle(struct alps_fields *f, unsigned char *p,
      struct psmouse *psmouse)
 {
  f->first_mp = !!(p[4] & 0x40);
@@ -558,9 +558,11 @@ static void alps_decode_pinnacle(struct alps_fields *f, unsigned char *p,
  f->pressure = p[5] & 0x7f;
 
  alps_decode_buttons_v3(f, p);
+
+ return 0;
 }
 
-static void alps_decode_rushmore(struct alps_fields *f, unsigned char *p,
+static int alps_decode_rushmore(struct alps_fields *f, unsigned char *p,
      struct psmouse *psmouse)
 {
  alps_decode_pinnacle(f, p, psmouse);
@@ -570,9 +572,11 @@ static void alps_decode_rushmore(struct alps_fields *f, unsigned char *p,
  f->fingers = max((p[5] & 0x3), ((p[5] >> 2) & 0x3)) + 1;
  f->x_map |= (p[5] & 0x10) << 11;
  f->y_map |= (p[5] & 0x20) << 6;
+
+ return 0;
 }
 
-static void alps_decode_dolphin(struct alps_fields *f, unsigned char *p,
+static int alps_decode_dolphin(struct alps_fields *f, unsigned char *p,
     struct psmouse *psmouse)
 {
  u64 palm_data = 0;
@@ -605,6 +609,8 @@ static void alps_decode_dolphin(struct alps_fields *f, unsigned char *p,
   f->x_map = (palm_data >> priv->y_bits) &
       (BIT(priv->x_bits) - 1);
  }
+
+ return 0;
 }
 
 static void alps_process_touchpad_packet_v3_v5(struct psmouse *psmouse)
diff --git a/drivers/input/mouse/alps.h b/drivers/input/mouse/alps.h
index ee841e5..17e3ae3 100644
--- a/drivers/input/mouse/alps.h
+++ b/drivers/input/mouse/alps.h
@@ -157,7 +157,7 @@ struct alps_data {
 
  int (*hw_init)(struct psmouse *psmouse);
  void (*process_packet)(struct psmouse *psmouse);
- void (*decode_fields)(struct alps_fields *f, unsigned char *p,
+ int (*decode_fields)(struct alps_fields *f, unsigned char *p,
          struct psmouse *psmouse);
  void (*set_abs_params)(struct alps_data *priv, struct input_dev *dev1);
 
-- 
1.7.1